test/transcoding/SPV_KHR_integer_dot_product-nonsat.ll

```diff
@@ -0,0 +1,292 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -spirv-text --spirv-ext=+SPV_KHR_integer_dot_product -o %t.txt
```

Comment:
What will happen if the extension isn't enabled?

Reply:
The translator will hit an assert related to `isAllowedToUseExtension`, as the translator starts with an empty `ExtensionsStatusMap` if no `--spirv-ext` is given.  So it will refuse to produce SPIR-V in that case.

Reply:
May we use:
`; RUN: not --crash llvm-spirv %t.bc -o %t.spv | FileCheck %s --check-prefix=CHECK-ERROR`
to test this case?

Reply:
Done.  The translator currently just asserts as I mentioned, which is not very pretty.  I will try to improve that in a separate pull request, once this has landed.

Reply:
Yeah, I believe it's a general problem with SPIR-V friendly LLVM IR. Usual pattern like:
 IF something_in_llvm_ir
  AND IF allowed_extension
   THEN generate_some_spirv
isn't applicable in this flow.

Reply:
The CI failures show that we cannot check for this in all possible builds at the moment: for a build without asserts, we cannot check for any error messages.

Reply:
Turning the assert into a proper error seemed easy actually, so I've done it as part of this PR.